src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ExecuteTaskReconfigurationServiceIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+        doReturn(taskResource1).when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource1);
+
+        TaskResource taskResource2 = taskResources.get(1);
+        assert taskResource2 != null;
+        taskResource2.setTitle("title1");
+        doThrow(new OptimisticLockException("locked"))
+            .when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource2);
+
+        TaskResource taskResource3 = taskResources.get(2);
+        assert taskResource3 != null;
+        taskResource3.setTitle("title1");
+        doReturn(taskResource3).when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource3);
+
+        TaskConfigurationResults results = new TaskConfigurationResults(
+            emptyMap(),
+            configurationDmnResponse(),
+            permissionsResponse()
+        );
+        doReturn(results).when(caseConfigurationProviderService)
+            .getCaseRelatedConfiguration(anyString(), anyMap(), anyBoolean());
+        List<TaskFilter<?>> taskFilters = createReconfigureTaskFilters();
+        TaskOperationRequest request = new TaskOperationRequest(
+            TaskOperation.builder()
+                .type(TaskOperationType.EXECUTE_RECONFIGURE)
+                .retryWindowHours(1L)
+                .maxTimeLimit(30)
+                .runId("")
+                .build(),
+            taskFilters
+        );
+
+        try {
+            executeTaskReconfigurationService.performOperation(request);
+        } catch (TaskExecuteReconfigurationException e) {
+            log.error("TaskExecuteReconfigurationException Exception {}", e.getMessage());
+        }
+
+        //doInTransaction(() -> executeTaskReconfigurationService.performOperation(request));
+        verify(cftTaskDatabaseService, times(2)).saveTask(any(TaskResource.class));
+        // 6 times because the taskResource2 will be retried
+        verify(taskReconfigurationHelper, times(6)).reconfigureTaskResource(anyString());
+
+        final TaskResource taskResource1AfterReconfigure =
+            taskResourceRepository.getByTaskId(taskResource1.getTaskId()).orElse(null);
+        final TaskResource taskResource2AfterReconfigure =
+            taskResourceRepository.getByTaskId(taskResource2.getTaskId()).orElse(null);
+        final TaskResource taskResource3AfterReconfigure =
+            taskResourceRepository.getByTaskId(taskResource3.getTaskId()).orElse(null);
+
+        assertAll(
+            () -> {
+                assert taskResource1AfterReconfigure != null;
+                assertEquals("title1", taskResource1AfterReconfigure.getTitle());
+            },
+            () -> {
+                assert taskResource2AfterReconfigure != null;
+                assertEquals("title", taskResource2AfterReconfigure.getTitle());
+            },
+            () -> {
+                assert taskResource3AfterReconfigure != null;
+                assertEquals("title1", taskResource3AfterReconfigure.getTitle());
+            }
+        );
```

Comment:
So if I understand correctly taskResource2 is Reconfigured and the others are not?

Reply:
So on line Sorry if I misunderstand.  In line: https://github.com/hmcts/wa-task-management-api/pull/1343/commits/04caaced82e8d7455c84dc2d57a3bd70bcee3d4a#diff-6f7dd1eb257b8f89f3eda999ac336872c341871863100b72d1d38f4ec6ddd522R104

you set this value to `title1` and then it is changed to set to `title`.  Does that mean it was reconfigured?